src/lib.rs

```diff
@@ -52,4 +52,8 @@ pub mod stream;
 pub mod sync;
 pub mod task;
 
+#[cfg_attr(feature = "docs", doc(cfg(unstable)))]
+#[cfg(feature = "unstable")]
+pub mod pin;
```

Comment:
Also, I think it's better to re-export the `std::pin` module itself instead of defining our own module. `std::pin` module documentation is great (and proper use of Pin API without `std::pin` module documentation is difficult).

Reply:
I've added a link to `std::pin`'s docs. I don't think we should directly re-export `std::pin` in full because we may want to consider adding more things to it in the submodule in the future that currently aren't part of std. This should hopefully make for a good enough middle ground.

Reply:
Sounds good to me.